Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -155,9 +155,16 @@ public async Task Push(Layer layer, string name, Action<string> logProgressMessa
         return false;
     }
 
+    private static HttpClient _client = CreateClient();
+
     private static HttpClient GetClient()
     {
-        var clientHandler = new AuthHandshakeMessageHandler(new HttpClientHandler() { UseDefaultCredentials = true });
+        return _client;
+    }
+
+    private static HttpClient CreateClient()
```

Comment:
I would have expected GetClient() to return some cached HttpClient. Is it the AuthHandshakeMessageHandler that's causing it to be long-lived?